src/InvestmentManager.sol

```diff
@@ -365,8 +365,13 @@ contract InvestmentManager is Auth {
         address _currency = poolManager.currencyIdToAddress(currency);
         address liquidityPool = poolManager.getLiquidityPool(poolId, trancheId, _currency);
         require(liquidityPool != address(0), "InvestmentManager/tranche-does-not-exist");
-
         LPValues storage lpValues = orderbook[liquidityPool][recipient];
+
+        // Sanity check that there is indeed an outstanding order for this recipient
+        // before transferring funds to the user escrow
+        require(lpValues.remainingRedeemOrder != 0, "InvestmentManager/no-outstanding-order");
```

Comment:
here we might get an issue, if the tx order is not sequential.
Example:
- order is fullfilled within 2 epochs:
- the second tx from the second epoch comes in first & sets the remaining order to 0.
-  when the tx from the first epoch execution arrives - it fails, since remainingRedeemOrder was already set to 0 by the second epoch tx.

Reply:
This scenario would be a good test case